lib/flags.ml

```diff
@@ -175,6 +175,12 @@ let make_polymorphic_flag b =
 let program_mode = ref false
 let is_program_mode () = !program_mode
 
+(** [function_mode] tells that Function mode has been activated, either
+    globally via [Set Program] or locally via the Function command. *)
+
+let function_mode = ref false
+let is_function_mode () = !function_mode
```

Comment:
What do you prefer: 
I can move that code in Pretyping module and export it to the whole word or add an optional argument to the full chain from pretype_type to ise_pretype_gen. 
I must admit I would prefer the first one (less code to write) but I can do both